tst/com/amazon/ion/benchmark/DataGeneratorTest.java

```diff
@@ -204,39 +206,34 @@ public void testSizeOfGeneratedData() throws Exception {
      * @throws Exception if error occur
      */
     @Test
-    public void testViolation() throws Exception {
-        Map <String, Object> optionsMap = Main.parseArguments("generate", "--data-size", "5000", "--format", "ion_text", "--input-ion-schema", "./tst/com/amazon/ion/benchmark/testStruct.isl", "test8.ion");
-        String inputFilePath = optionsMap.get("--input-ion-schema").toString();
-        Path classSchemaFile = Paths.get(inputFilePath);
-        OUTPUT_FILE = optionsMap.get("<output_file>").toString();
-        try (
-                IonReader readerInput = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(inputFilePath)));
-                IonReader reader = DataGeneratorTest.executeAndRead(optionsMap);
-                Stream<String> lines = Files.lines(classSchemaFile)
-        ) {
-            // Get the name of Ion Schema.
-            IonDatagram schema = ReadGeneralConstraints.LOADER.load(readerInput);
-            String ionSchemaName = null;
-            for (int i = 0; i < schema.size(); i++) {
-                IonValue schemaValue = schema.get(i);
-                if (schemaValue.getType().equals(IonType.STRUCT) && schemaValue.getTypeAnnotations()[0].equals(IonSchemaUtilities.KEYWORD_TYPE)) {
-                    IonStruct constraintStruct = (IonStruct) schemaValue;
-                    ionSchemaName = constraintStruct.get(IonSchemaUtilities.KEYWORD_NAME).toString();
-                    break;
+    public void testViolationOfIonStruct() throws Exception {
+        for (int index = 0; index < SCHEMA_FILES.size(); index++) {
```

Comment:
This is great to see that you're reusing this code for all of your test schemas. The only problem with this technique is that if one of the files starts failing, it's not immediately clear which one because all of them are tested in the same test method. Therefore, I recommend moving all of the code inside the loop into a reusable method that accepts the schema file, and creating a separate `@Test` method for each file, which will simply call into the method. That way it's always clear which file(s) caused test failures and it's easy to target just the tests for the failing files for debugging.

Another option is [parameterized testing](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests), which is great way to achieve this using JUnit.